### DISCUSSION TITLE
How to get data from a flow variable?

### DISCUSSION THREAD

**Discussion by SkateWarp** [zachowj/node-red-contrib-home-assistant-websocket#477] [Q&A] [answered]
Hello! 
I am having a problem, I can't find the right way to read a flow variable, and pass it inside of call service node, in data.

![imagen](https://user-images.githubusercontent.com/34940433/135474528-b217b037-3fa3-4804-a4dd-846010fbcd9f.png)
How i'm trying to read the data (as far I read the docs)

![imagen](https://user-images.githubusercontent.com/34940433/135474733-79590471-0ab1-4bed-b28e-4666534bd95d.png)
Node

![imagen](https://user-images.githubusercontent.com/34940433/135474885-264c8499-cce0-402a-b88a-7ca1c500015b.png)
The content of the flow variable 

Few days ago, it worked okay.

I'm trying to restore the old volume level of a Nest Audio, after being used to play a mp3 file. 
Thank you in advance.

**Comment by zachowj** [comment, marked as answer]
I believe mustache templates are only able to pull from the default storage context. You could change the data field type to JSONata and do it this way.

```js
{
   "volume_level": $flowContext("data.attributes.volume_level", "file")
}
```

It would work as you have it if the flow data was written the default context and not the file context.

**Reply by SkateWarp** [reply]
I didn't know that mustache templates only read data (by default) from the memory only.
Anyhow, it worked! Thank you! :)
May I suggest something? It'd be handy having an example of reading a file (flow or global), in the docs. 👍
